rlberry/utils/check_agent.py

```diff
@@ -0,0 +1,202 @@
+from rlberry.envs import Chain
+from rlberry.envs.benchmarks.ball_exploration import PBall2D
+from rlberry.manager import AgentManager
+import numpy as np
+from rlberry.seeding import set_external_seed
+import tempfile
+import os
+
+SEED = 42
+
+
+def check_agent_manager(Agent, continuous_state=False):
+    """
+    Check that the agent is compatible with :class:`~rlberry.manager.AgentManager`.
+
+    Parameters
+    ----------
+
+    Agent: rlberry agent module
+        Agent class that we want to test.
+
+    continuous_state: bool, default=False
+        whether we test on a discrete state environment or a continuous one.
+    """
+    if continuous_state:
+        env_ctor = PBall2D
+    else:
+        env_ctor = Chain
+    env_kwargs = {}
+    try:
+        agent = AgentManager(
+            Agent, (env_ctor, env_kwargs), fit_budget=5, n_fit=1, seed=SEED
+        )
+        agent.fit()
+    except Exception as exc:
+        raise RuntimeError("Agent not compatible with Agent Manager") from exc
+
+    return agent
+
+
+def check_agent_almost_equal(state, agent1, agent2, n_checks=5):
+    """
+    Check that two agents are almost equal in the sense that they give the same
+    actions n_checks times on a given state.
+
+
+    Parameters
+    ----------
+
+    state: state
+        environment state
+    agent1: rlberry agent
+        first agent to be compared
+    agent2: rlberry agent
+        second agent to be compared
+    n_checks : int, default=5
+        number of checks to do.
+    """
+    result = True
+    set_external_seed(SEED)
```

Comment:
I actually tried safe_reseed and for some reason it does not work with PPOAgent, AVECPPOAgent and REINFORCEAgent. on the other hand, there is no problem with what I did if we suppose that the agents have the same seed (if they don't have they are not almost_equal).

Reply:
For safe_reseed I agree and also I need the set_external_seed even though the seed has already been specified in AgentManager, this is due to torch using its own seed. Maybe we could do something to merge all the seeds ? I don't know, @omardrwch what do you think ?

For agents with different RNG, yes but I don't know how to test that two agents are the same because it depends on how they are describe (weights for neural networks, nulber of pulls for bandits...). For now I will keep it like that because in fact I use the test only with AgentManagers with the same seed and I don't need it for any other application but that may change in the futur.

Reply:
What do you mean by "merge all seeds"? In the library, we take the following approach:

* Each agent or environment has its own random number generator (rng) and, consequently, its own seed. This is very important to avoid global seeds, which are horrible for multiprocessing/threading (where we can't control how global variables are handled). Global seeds are also horrible in general, in the sense that if agents A and B rely on a global seed, running A then B is different from running B then A (which is precisely our issue with pytorch here).
* Not all libraries handle seeds that way, that is, they use global seeds (as pytorch, but not jax :D), and that is why we have the `set_external_seed` function. **This function is only supposed to be called one time in the beginning of each process** to avoid weird behaviors with respect to reproducibility and to make sure that all seeds used in a single process are independent.  By the way, since we're calling it  in `check_agent`, I believe this should raise a **big warning** to the user.

Reply:
Ok, thanks this makes things clearer for me. Maybe I can include a message that say "check that reproducible if a global seed is given" in the docstring.